dbms/src/DataStreams/SortHelper.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/SortHelper.h>
+
+namespace DB::SortHelper
+{
+void removeConstantsFromBlock(Block & block)
+{
+    size_t columns = block.columns();
+    size_t i = 0;
+    while (i < columns)
+    {
+        if (block.getByPosition(i).column->isColumnConst())
+        {
+            block.erase(i);
+            --columns;
+        }
+        else
+            ++i;
+    }
+}
+
+void removeConstantsFromSortDescription(const Block & header, SortDescription & description)
+{
+    description.erase(
```

Comment:
Although not releated to your newly added code, but there maybe a potential problem that the const column in `header` and `block` is not the same, I think we need to record the const column in `header` as the meta data, then only remove the column recoreded when call `removeConstantsFromBlock`?

Reply:
I prefer to modify in another pr..